### ISSUE TITLE
Constant visibility false positive on non-class constants

### ISSUE THREAD

**Issue by dereuromark** [squizlabs/PHP_CodeSniffer#2615]
```
 3 | WARNING | Visibility must be declared on all constants if your
   |         | project supports PHP 7.1 or later
```

But the code is
```php
<?php

const APPLICATION_ENV = 'development';
```

so a normal classless file, as such visibility for such a const cannot be given, can it?

**Comment by jrfnl** [comment]
Confirmed. This is a bug.

**Comment by Beakerboy** [comment]
What Is the resolution on this going to look like? I have a project that is advertised as supporting php 7.0+. Class constants just started being flagged with this today on Travis-ci. Since visibility on constants didn’t appear until 7.1, do I need to turn this check off in my coding standards file?

**Comment by GMTA** [comment]
For completeness, the sniff is `PSR12.Properties.ConstantVisibility.NotFound`.

**Comment by gsherwood** [comment]
Thanks for reporting this. Fix will be in 3.5.1, which will be released when I've fixed the other PSR-12 issues that have come through. Check out the milestone for the list.

**Comment by vaso123** [comment]
Not really fixed. It is also in: PHP_CodeSniffer version 3.5.8 (stable) by Squiz (http://www.squiz.net) with PHPStorm. I am using PSR-12

![const](https://user-images.githubusercontent.com/2932257/97308205-e9995800-1860-11eb-80e0-4afa8b021241.png)

**Comment by jrfnl** [comment]
@vaso123 I think you misunderstand the earlier reported issue. Constant visibility was previously - incorrectly - demanded for non-class constants. That bug has been fixed and can no longer be reproduced.

I suspect you don't want to use class constant visibility as your project needs to support older PHP versions. Is that correct ?
In that case, you will need to ignore the rule from within your custom ruleset or from the command line.
This is not a bug. PSR12 demands visibility for class constants. You can choose not to abide by it.

**Comment by vaso123** [comment]
Ok, got it, you are right, sorry.
